control-plane/agents/core/src/volume/specs.rs

```diff
@@ -413,17 +413,18 @@ impl ResourceSpecsLocked {
         if let Some(volume) = &volume {
             SpecOperations::start_destroy(volume, registry, false, mode).await?;
 
-            let mut first_error = Ok(());
             let nexuses = self.get_volume_nexuses(&request.uuid);
             for nexus in nexuses {
                 let nexus = nexus.lock().deref().clone();
                 if let Err(error) = self
-                    .destroy_nexus(registry, &DestroyNexus::from(nexus), true, mode)
+                    .destroy_nexus(registry, &DestroyNexus::from(nexus.clone()), true, mode)
```

Comment:
Is it worth adding a note above the `destroy_volume` function that it success even if it can't delete the data replicas? (might be relevant for secure erase)

Reply:
Good suggestion, done.